Overhaul RA installed content.

pchote opened the pull request:
This overhauls the RA asset extraction and downloads with a few goals in mind:
- Aftermath and C&C Desert assets are now managed by the content installer system.  This removes a bunch of binary assets from the repo and cuts almost 1MB from the windows installer download size (according to Appveyor).
- Downloaded asset packages provide identical content to the data installed from disc.
- lores.mix gives us a full set of low resolution icons for beacons and the observer interface.

Depends on #11536.
Closes #11537.

pchote committed:
Load aftermath assets from content dir.

pchote committed:
Move RA assets to v2 dir for backwards compatibility.

pchote committed:
Load C&C desert assets from content dir.

pchote committed:
Extract only the redalert.mix contents that we need.

pchote committed:
Load beacon posters from lores.mix.

pchote committed:
Update asset downloads.

pchote commented:
backwards compatibility (new playtest <-> last release and new bleed <-> older bleed) is going to be a problem here, and the migrations code won't be able to help.  I think our only viable option will be to namespace the assets by putting them in a different directory.

obrakmann commented:
In Ruby On Rails, migrations used to have an `up()` and a `down()` method, for up- and downgrading respectively, obviously. Would that help here? We would need a way to store the versions of mods that were last loaded/installed, though. But I think that would be helpful in the long run, anyway, once we allow in-game mod installation where people can jump back and forth between mod versions with just a couple of mouse clicks.

pchote commented:
I don't think that's worth the effort, and honestly would probably give up on finishing this rework if that became a requirement.  The main reason this sucks so badly is that our old package setup is full of years of hacks and technical debt.  Once everyone migrates to a consistent and non-mangled set of packages then future changes should be limited to simple additions (if we need assets in mix files that aren't included).

obrakmann commented:
No, it's cool, I'm fine with an "upgrades only"-policy. The mod browser should enforce that then, though, otherwise we'd probably get flooded with bug reports.

pchote commented:
Yup, that's why I added the changes requested tag.  I'm going to wait for #11536 before I touch this again.

pchote committed:
Revert "Add Migrations GlobalModData."

This reverts commit c204c10f5e3a487105f21cac880799a9db1b9011.

pchote commented:
Updated.

This now fixes #11665 by completely removing the now-redundant migrations feature.


I have split #11741 to its own ticket because addressing it here would add several hundred more lines to mod.yaml

obrakmann commented on mods/ra/mod.yaml:
```diff
@@ -240,10 +240,12 @@ ModContent:
 			TestFiles: ^Content/ra/v2/expand/expand2.mix, ^Content/ra/v2/expand/hires1.mix, ^Content/ra/v2/expand/lores1.mix, ^Content/ra/v2/expand/chrotnk1.aud, ^Content/ra/v2/expand/fixit1.aud, ^Content/ra/v2/expand/jburn1.aud, ^Content/ra/v2/expand/jchrge1.aud, ^Content/ra/v2/expand/jcrisp1.aud, ^Content/ra/v2/expand/jdance1.aud, ^Content/ra/v2/expand/jjuice1.aud, ^Content/ra/v2/expand/jjump1.aud, ^Con ...
 			Sources: aftermath, aftermath-linux, tfd, ra-origin
 			Required: true
+			Download: aftermath
```
Aftermath wasn't released as freeware, was it?

pchote replied:
Not explicitly, no.  Nobody seems to care about that fact though and it is far too late for us to remove our dependency on its assets.

obrakmann replied:
Well, we're shipping it already, this is just changing the mode of delivery, so to speak. Guess we'll have to live with it.

obrakmann commented:
Looks fine, :+1:

Mailaender commented on mods/ra/mod.yaml:
```diff
@@ -8,15 +8,16 @@ RequiresMods:
 	$cnc: cnc
 	./mods/common: common
 	~main.mix
-	~redalert.mix
```
This makes the game not detect manually copied installations and attempt to download redundant files.

Mailaender replied:
Guess this is a trade-off for better performance. https://github.com/OpenRA/OpenRA/issues/11741

pchote replied:
one of the pr goals is to guarantee a standard set of installed assets.  Calling this a feature might be a bit far, but it's certainly not an unwanted side effect.

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8b66ea95d7e16c270ccbf9b23fee73fed56e52b6...68a508305aff53f9e01854d6566b8327480b7803)

reaperrr commented:
This regressed the RA shellmap slightly, there's now a tile error on th river near the right soviet base.

Probably a missing file that somehow didn't get included in the RA desert content download.
